Use the latest trustymail_reporter and gatherer Docker images

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the Docker composition to use the [latest `cisagov/gatherer`](cisagov/gatherer#66) and [latest `cisagov/trustymail_reporter`](cisagov/trustymail_reporter#59) Docker images.

While this PR awaited review cisagov/trustymail_reporter#60 came to pass, so I have included it here as well.

## 💭 Motivation and context ##

These latest Docker images are built from the latest Debian-based Python 3.10 Docker images.

## 🧪 Testing ##

All automated tests pass.  I used these new Docker images in this weekend's BOD 18-01 reporting run.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Add a tag.

jsf9k committed:
Bump version from 1.3.37 to 1.3.38

jsf9k committed:
Use the latest gatherer and trustymail_reporter Docker images

mcdonnnj commented on docker-compose.yml:
```diff
@@ -50,7 +50,7 @@ services:
       - ./output:/home/saver/shared
   trustymail_report:
-    image: 'cisagov/trustymail_reporter:1.4.1'
+    image: 'cisagov/trustymail_reporter:1.4.2'
```
Given https://github.com/cisagov/trustymail_reporter/pull/60 should this be updated?

```suggestion
    image: 'cisagov/trustymail_reporter:1.4.3'
```

jsf9k replied:
Yes.  I was waiting for cisagov/trustymail_reporter#60 to get approved and merged before making that change.

mcdonnnj reviewed (approved):
Makes sense. I had one question about image versions.

jsf9k committed:
Use the very latest trustymail_reporter Docker image

This Docker image was updated while this PR awaited review, so we may as well make use of it.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>
